test/src/main/java/org/apache/accumulo/test/compaction/CompactionPriorityQueueMetricsIT.java

```diff
@@ -0,0 +1,397 @@
+      Map<String,String> sysProps = Map.of(TestStatsDRegistryFactory.SERVER_HOST, "127.0.0.1",
+          TestStatsDRegistryFactory.SERVER_PORT, Integer.toString(sink.getPort()));
+      cfg.setSystemProperties(sysProps);
+    }
+  }
+
+  private String writeData(String file, AccumuloConfiguration aconf, int s, int e)
+      throws Exception {
+    FileSystem fs = getCluster().getFileSystem();
+    String filename = file + RFile.EXTENSION;
+    try (FileSKVWriter writer = FileOperations.getInstance().newWriterBuilder()
+        .forFile(UnreferencedTabletFile.of(fs, new Path(filename)), fs, fs.getConf(),
+            NoCryptoServiceFactory.NONE)
+        .withTableConfiguration(aconf).build()) {
+      writer.startDefaultLocalityGroup();
+      for (int i = s; i <= e; i++) {
+        writer.append(new Key(new Text(row(i))), new Value(Integer.toString(i)));
+      }
+    }
+
+    return hash(filename);
+  }
+
+  private void addSplits(AccumuloClient client, String tableName, String splitString)
+      throws Exception {
+    SortedSet<Text> splits = new TreeSet<>();
+    for (String split : splitString.split(" ")) {
+      splits.add(new Text(split));
+    }
+    client.tableOperations().addSplits(tableName, splits);
+  }
+
+  private void verifyData(AccumuloClient client, String table, int start, int end, boolean setTime)
+      throws Exception {
+    try (Scanner scanner = client.createScanner(table, Authorizations.EMPTY)) {
+
+      Iterator<Map.Entry<Key,Value>> iter = scanner.iterator();
+
+      for (int i = start; i <= end; i++) {
+        if (!iter.hasNext()) {
+          throw new Exception("row " + i + " not found");
+        }
+
+        Map.Entry<Key,Value> entry = iter.next();
+
+        String row = String.format("%04d", i);
+
+        if (!entry.getKey().getRow().equals(new Text(row))) {
+          throw new Exception("unexpected row " + entry.getKey() + " " + i);
+        }
+
+        if (Integer.parseInt(entry.getValue().toString()) != i) {
+          throw new Exception("unexpected value " + entry + " " + i);
+        }
+
+        if (setTime) {
+          assertEquals(1L, entry.getKey().getTimestamp());
+        }
+      }
+
+      if (iter.hasNext()) {
+        throw new Exception("found more than expected " + iter.next());
+      }
+    }
+  }
+
+  @SuppressFBWarnings(value = {"PATH_TRAVERSAL_IN", "WEAK_MESSAGE_DIGEST_SHA1"},
+      justification = "path provided by test; sha-1 is okay for test")
+  private String hash(String filename) {
+    try {
+      byte[] data = Files.readAllBytes(Paths.get(filename.replaceFirst("^file:", "")));
+      byte[] hash = MessageDigest.getInstance("SHA1").digest(data);
+      return new BigInteger(1, hash).toString(16);
+    } catch (IOException | NoSuchAlgorithmException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static String row(int r) {
+    return String.format("%04d", r);
+  }
+
+  @Test
+  public void testQueueMetrics() throws Exception {
```

Comment:
Would be nice to test metrics from multiple queues, just to ensure the data for queue is independent.  Could be a follow in issue.